server/php-master.cpp

```diff
@@ -763,12 +763,13 @@ int run_worker(WorkerType worker_type) {
     switch (worker_type) {
       case WorkerType::http_worker:
         run_mode = RunMode::http_worker;
+        prctl(PR_SET_PDEATHSIG, SIGKILL); // TODO: or SIGTERM
         break;
       case WorkerType::job_worker:
         run_mode = RunMode::job_worker;
+        prctl(PR_SET_PDEATHSIG, SIGTERM);
```

Comment:
I'm afraid, but the master process should be terminated last and after him no processes should left.
1) Some stuff are destroyed on master process shutdown 
2) Also, if a job worker is not terminated, it continues working as a zombie processes

I suggest using `prctl(PR_SET_PDEATHSIG, SIGKILL);` for all child and processing termination from master process.

Reply:
Agree, fixed